src/SmartComponents/DriftPage/DriftTable/DriftTable.js

```diff
@@ -215,19 +215,78 @@ class DriftTable extends Component {
         return row;
     }
 
-    renderSortButton(sort) {
+    renderSortButton(sortType, sort) {
         let sortIcon;
 
         if (sort === 'asc') {
-            sortIcon = <AngleUpIcon className="pointer active-blue" onClick={ () => this.props.toggleFactSort('desc') }/>;
+            sortIcon = <ArrowUpIcon className="pointer active-blue" onClick={ () => this.toggleSort(sortType, sort) }/>;
         }
         else if (sort === 'desc') {
-            sortIcon = <AngleDownIcon className="pointer active-blue" onClick={ () => this.props.toggleFactSort('asc') }/>;
+            sortIcon = <ArrowDownIcon className="pointer active-blue" onClick={ () => this.toggleSort(sortType, sort) }/>;
+        }
+        else {
+            sortIcon = <ArrowsAltVIcon className="pointer" onClick={ () => this.toggleSort(sortType, sort) }/>;
         }
 
         return sortIcon;
     }
 
+    toggleSort(sortType, sort) {
+        if (sort === 'asc') {
+            if (sortType === 'fact') {
+                this.props.toggleFactSort('desc');
+            } else {
+                this.props.toggleStateSort('desc');
```

Comment:
If the `sort` that is passed in is `asc` why do we pass `desc` into the function? My intuition says it should be just `sort` that is passed in. What am I missing?

Reply:
So, if the current sort is ascending, and someone clicks it to change it to descending, I'm passing 'desc' to the action so we can update the sort in the reducer by simply doing this: https://github.com/RedHatInsights/drift-frontend/pull/131/files#diff-d585a54c07f675fb1ae7d1906b05671eR391

I can change it to put the logic you mention above into the reducer instead, if you think that makes more sense.